### PR TITLE
JDF-168 - Contributing Guide

### PR DESCRIPTION
**Pull request by rafabene**
Contributing guide focused on the contributor perspective. So it does not include merging and release procedures

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,4 +1,158 @@
...
+
+2. Clone your fork (`git@github.com:<your-username>/jboss-as-quickstart.git`).
+
+3. Add an `upstream` remote (`git remote add upstream git@github.com:jboss-jdf/jboss-as-quickstart.git`).
+
+4. Get the latest changes from upstream (e.g. `git pull upstream master`).
+
+5. Create a new topic branch to contain your feature, change, or fix (`git checkout -b <topic-branch-name>`).
+
+6. Make sure that your changes follow the General Guide Lines.
```
Comment (sgilda) [inline CONTRIBUTING.md]: Make sure that your changes follow the General Guidelines below.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,4 +1,158 @@
...
+
+1. [Fork](http://help.github.com/fork-a-repo/) the project.
+
+2. Clone your fork (`git@github.com:<your-username>/jboss-as-quickstart.git`).
+
+3. Add an `upstream` remote (`git remote add upstream git@github.com:jboss-jdf/jboss-as-quickstart.git`).
+
+4. Get the latest changes from upstream (e.g. `git pull upstream master`).
+
+5. Create a new topic branch to contain your feature, change, or fix (`git checkout -b <topic-branch-name>`).
```
Comment (sgilda) [inline CONTRIBUTING.md:27]: Does the checkout need to specify upstream/master? In otherwords `git checkout -b <topic-branch-name> upstream/master`
I'm not a git expert, so I don't know. I always add that. :-)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,4 +1,158 @@
...
+
+* The quickstart project or folder name should match the quickstart name. Each sample project should have a unique name, allowing easy identification by users and developers.
+
+* If you create a quickstart that uses a database table, make sure the name you use for the table is unique across all quickstarts. 
+
+* The project must follow the structure used by existing quickstarts such as [numberguess](https://github.com/jboss-jdf/jboss-as-quickstart/tree/master/numberguess). A good starting point would be to copy the  `numberguess` project.
+
+* The sample project should be importable into JBoss Developer Studio/JBoss Tools and be deployable from there.
+
+* Maven POMs must be used. No other build system allowed unless the purpose of the quickstart is to show another build system in use. If using maven it should:
```
Comment (sgilda) [inline CONTRIBUTING.md]: s/maven/Maven
s/allowed/is allowed
- Maven POMs must be used. No other build system is allowed unless the purpose of the quickstart is to show another build system in use. If using Maven it should:

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,4 +1,158 @@
...
+
+
+Kitchensink variants
+--------------------
+
+  There are multiple quickstarts based on the kitchensink example.  Each showcases different technologies and techniques including pure EE6, JSF, HTML5, and GWT.  
+
+  If you wish to contribute a kitchensink variant is it important that you follow the look and feel of the original so that useful comparisons can be made.  This does not mean that variants can not expand, and showcase additional functionality.  Multiple variants already do that.  These include mobile interfaces, push updates, and more.
+
+  Below are rules for the l&f of the variants:
```
Comment (sgilda) [inline CONTRIBUTING.md]: I would expand "l&f " to "look and feel". I assume that's what it means? ;-)

**Comment by sgilda** [comment]
Other than the few minor comments, this looks good to me!

File: CONTRIBUTING.md (hunk 2 above)
Comment (rafabene) [inline CONTRIBUTING.md:27]: There's no need. If you don't specify, you will create the new branch from wherever the HEAD is

**Comment by sgilda** [comment]
This looks good to me. 
@pmuir, if you're good with this, I can merge.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,4 +1,158 @@
...
+
+- To allow examples to be copied by developers and used as the basis for their own projects.
+
+
+Basic Steps
+-----------
+
+To contribute with Quickstarts, clone your own fork instead of cloning the main Quickstarts repository, commit your work on topic branches and make pull requests. In detail:
+
+1. [Fork](http://help.github.com/fork-a-repo/) the project.
```
Comment (pmuir) [inline CONTRIBUTING.md]: You could include this handy link - https://github.com/jboss-jdf/jboss-as-quickstart/fork_select

**Comment by pmuir** [comment]
@sgilda looks good to me.

**Comment by sgilda** [comment]
Thanks Pete.
Rafael, let me know when you've added the link Pete mentioned above and I'll merge.

File: CONTRIBUTING.md (hunk 5 above)
Comment (rafabene) [inline CONTRIBUTING.md]: Cool!!! I'll add that to all other Contribute guides

**Comment by sgilda** [comment]
Merged.

**Pull request closed without merging**